### DISCUSSION TITLE
not compile

### DISCUSSION THREAD

**Discussion by micropythondevel** [micropython/micropython#15614] [RP2040 / Pico]
make V=1 -j -C ports/rp2 BOARD=PICO USER_C_MODULES=../../lib/lv_bindings/bindings.cmake                                                            ✔  20m 10s  

[sudo] contraseña para icelon: 
make: se entra en el directorio '/home/icelon/lvg/lv_micropython/ports/rp2'
[ -e build-PICO/Makefile ] || cmake -S . -B build-PICO -DPICO_BUILD_DOCS=0 -DMICROPY_BOARD=PICO -DMICROPY_BOARD_DIR=/home/icelon/lvg/lv_micropython/ports/rp2/boards/PICO -DUSER_C_MODULES=../../lib/lv_bindings/bindings.cmake
PICO_SDK_PATH is /home/icelon/lvg/lv_micropython/lib/pico-sdk
PICO platform is rp2040.
Build type is MinSizeRel
PICO target board is pico.
Using board configuration from /home/icelon/lvg/lv_micropython/lib/pico-sdk/src/boards/include/boards/pico.h
TinyUSB available at /home/icelon/lvg/lv_micropython/lib/tinyusb/src/portable/raspberrypi/rp2040; enabling build support for USB.
BTstack available at /home/icelon/lvg/lv_micropython/lib/pico-sdk/lib/btstack
cyw43-driver available at /home/icelon/lvg/lv_micropython/lib/pico-sdk/lib/cyw43-driver
Pico W Bluetooth build support available.
lwIP available at /home/icelon/lvg/lv_micropython/lib/lwip
mbedtls available at /home/icelon/lvg/lv_micropython/lib/pico-sdk/lib/mbedtls
Including User C Module(s) from ../../lib/lv_bindings/bindings.cmake
Found User C Module(s): usermod_lv_bindings, lvgl_interface
-- Configuring done (1.0s)
-- Generating done (0.3s)
-- Build files have been written to: /home/icelon/lvg/lv_micropython/ports/rp2/build-PICO
make -s -C build-PICO || (echo -e "See \033[1;31mhttps://github.com/micropython/micropython/wiki/Build-Troubleshooting\033[0m"; false)
[  0%] Creating directories for 'PioasmBuild'
[  0%] Creating directories for 'ELF2UF2Build'
[  0%] Building ASM object pico-sdk/src/rp2_common/boot_stage2/CMakeFiles/bs2_default.dir/compile_time_choice.S.obj
[  0%] No download step for 'PioasmBuild'
[  0%] No download step for 'ELF2UF2Build'
[  0%] Linking ASM executable bs2_default.elf
[  0%] No update step for 'ELF2UF2Build'
[  1%] No update step for 'PioasmBuild'
[  1%] Built target bs2_default
[  1%] No patch step for 'ELF2UF2Build'
[  1%] No patch step for 'PioasmBuild'
[  1%] Generating bs2_default.bin
[  1%] Performing configure step for 'ELF2UF2Build'
[  1%] Generating bs2_default_padded_checksummed.S
[  1%] Performing configure step for 'PioasmBuild'
loading initial cache file /home/icelon/lvg/lv_micropython/ports/rp2/build-PICO/pico-sdk/src/rp2_common/tinyusb/pioasm/tmp/PioasmBuild-cache-MinSizeRel.cmake
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

[  1%] Built target bs2_default_padded_checksummed_asm
-- The C compiler identification is GNU 14.1.1
-- The CXX compiler identification is GNU 14.1.1
-- The CXX compiler identification is GNU 14.1.1
-- Detecting C compiler ABI info
-- Detecting CXX compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.9s)
-- Generating done (0.0s)
-- Build files have been written to: /home/icelon/lvg/lv_micropython/ports/rp2/build-PICO/pioasm
[  1%] Performing build step for 'PioasmBuild'
[ 20%] Building CXX object CMakeFiles/pioasm.dir/gen/lexer.cpp.o
[ 30%] Building CXX object CMakeFiles/pioasm.dir/pio_assembler.cpp.o
[ 10%] Building CXX object CMakeFiles/pioasm.dir/main.cpp.o
[ 40%] Building CXX object CMakeFiles/pioasm.dir/pio_disassembler.cpp.o
[ 50%] Building CXX object CMakeFiles/pioasm.dir/gen/parser.cpp.o
[ 60%] Building CXX object CMakeFiles/pioasm.dir/c_sdk_output.cpp.o
[ 70%] Building CXX object CMakeFiles/pioasm.dir/hex_output.cpp.o
[ 80%] Building CXX object CMakeFiles/pioasm.dir/ada_output.cpp.o
[ 90%] Building CXX object CMakeFiles/pioasm.dir/python_output.cpp.o
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (1.5s)
-- Generating done (0.0s)
-- Build files have been written to: /home/icelon/lvg/lv_micropython/ports/rp2/build-PICO/elf2uf2
[  2%] Performing build step for 'ELF2UF2Build'
[ 50%] Building CXX object CMakeFiles/elf2uf2.dir/main.cpp.o
En el fichero incluido desde /home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:10:
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:16:25: error: ‘uint16_t’ no se declaró en este ámbito
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |                         ^~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:1:1: nota: ‘uint16_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
  +++ |+#include <cstdint>
    1 | /*
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:16:45: error: expected primary-expression before ‘sideset_bits’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |                                             ^~~~~~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:16:59: error: expected primary-expression before ‘bool’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |                                                           ^~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:17:53: error: ‘uint16_t’ no se ha declarado
   17 | extern "C" void disassemble(char *buf, int buf_len, uint16_t inst, uint sideset_bits, bool sideset_opt);
      |                                                     ^~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:17:103: error: ‘void disassemble(char*, int, int, uint, bool)’ redeclared as different kind of entity
   17 | extern "C" void disassemble(char *buf, int buf_len, uint16_t inst, uint sideset_bits, bool sideset_opt);
      |                                                                                                       ^
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:16:13: nota: previous declaration ‘std::string disassemble’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |             ^~~~~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:12:53: error: ‘uint16_t’ no se ha declarado
   12 | extern "C" void disassemble(char *buf, int buf_len, uint16_t inst, uint sideset_bits, bool sideset_opt) {
      |                                                     ^~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:12:103: error: ‘void disassemble(char*, int, int, uint, bool)’ redeclared as different kind of entity
   12 | extern "C" void disassemble(char *buf, int buf_len, uint16_t inst, uint sideset_bits, bool sideset_opt) {
      |                                                                                                       ^
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:16:13: nota: previous declaration ‘std::string disassemble’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |             ^~~~~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp: In function ‘void disassemble(char*, int, int, uint, bool)’:
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:13:33: error: no hay coincidencia para la llamada a ‘(std::string {aka std::__cxx11::basic_string<char>}) (int&, uint&, bool&)’
   13 |     if (buf_len) buf[disassemble(inst, sideset_bits, sideset_opt).copy(buf, buf_len - 1)] = 0;
      |                      ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp: En el ámbito global:
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:16:13: error: redefinición de ‘std::string disassemble’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits_including_opt, bool sideset_opt) {
      |             ^~~~~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.h:16:13: nota: ‘std::string disassemble’ previously declared here
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |             ^~~~~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:16:25: error: ‘uint16_t’ no se declaró en este ámbito
   16 | std::string disassemble(uint16_t inst, uint sideset_bits_including_opt, bool sideset_opt) {
      |                         ^~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:11:1: nota: ‘uint16_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   10 | #include "pio_disassembler.h"
  +++ |+#include <cstdint>
   11 | 
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:16:45: error: expected primary-expression before ‘sideset_bits_including_opt’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits_including_opt, bool sideset_opt) {
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/icelon/lvg/lv_micropython/lib/pico-sdk/tools/pioasm/pio_disassembler.cpp:16:73: error: expected primary-expression before ‘bool’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits_including_opt, bool sideset_opt) {
      |                                                                         ^~~~
make[6]: *** [CMakeFiles/pioasm.dir/build.make:104: CMakeFiles/pioasm.dir/pio_disassembler.cpp.o] Error 1
make[6]: *** Se espera a que terminen otras tareas....
[100%] Linking CXX executable elf2uf2
[100%] Built target elf2uf2
[  2%] No install step for 'ELF2UF2Build'
[  2%] Completed 'ELF2UF2Build'
[  2%] Built target ELF2UF2Build
make[5]: *** [CMakeFiles/Makefile2:83: CMakeFiles/pioasm.dir/all] Error 2
make[4]: *** [Makefile:91: all] Error 2
make[3]: *** [pico-sdk/src/rp2_common/tinyusb/CMakeFiles/PioasmBuild.dir/build.make:86: pico-sdk/src/rp2_common/tinyusb/pioasm/src/PioasmBuild-stamp/PioasmBuild-build] Error 2
make[2]: *** [CMakeFiles/Makefile2:1679: pico-sdk/src/rp2_common/tinyusb/CMakeFiles/PioasmBuild.dir/all] Error 2
make[2]: *** Se espera a que terminen otras tareas....
GEN /home/icelon/lvg/lv_micropython/ports/rp2/build-PICO/genhdr/mpversion.h
[  2%] Built target BUILD_VERSION_HEADER
make[1]: *** [Makefile:91: all] Error 2
See https://github.com/micropython/micropython/wiki/Build-Troubleshooting
make: *** [Makefile:46: all] Error 1
make: se sale del directorio '/home/icelon/lvg/lv_micropython/ports/rp2'

**Comment by Josverl** [comment]
You can use [GitHub Flavoured Markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/about-writing-and-formatting-on-github) to format your posts. In particular use ```py ... ``` around your code snippets to get syntax highlighting, see the section below.

https://github.com/orgs/micropython/discussions/9111
